src/app/user/register/register.component.ts

```diff
@@ -9,38 +11,126 @@ import { Router } from '@angular/router';
 })
 
 export class RegisterComponent implements OnInit {
-  hide = true;
+  hidePasswords = true;
   registerError = false;
+  errorMessage = '';
+  registerErrorMessage = '';
+  testvar = false;
 
-  constructor(public auth: AngularFireAuth, public router: Router) { }
+  registerForm: FormGroup;
+  userEmailControl: FormControl;
+
+  userPasswordsGroup: FormGroup;
+  userPasswordControl: FormControl;
+  userPasswordConfirmControl: FormControl;
+
+  confirmPasswordsMatch = new ConfirmPasswordsMatch();
+
+  constructor(public auth: AngularFireAuth, public router: Router) {
+    this.testvar = false;
+
+    // User Email
+    this.userEmailControl = new FormControl('', [
+      Validators.required,
+      Validators.email
+    ]);
+
+    // User Password
+    this.userPasswordControl = new FormControl('', [
+      Validators.required,
+    ]);
+
+    // User Password confirmation
+    this.userPasswordConfirmControl = new FormControl('', [
+      Validators.required,
+    ]);
+
+    // Passwords group
+    this.userPasswordsGroup = new FormGroup({
+      userPasswordControl: this.userPasswordControl,
+      userPasswordConfirmControl: this.userPasswordConfirmControl
+    }, 
+    { validators: this.passwordsValid.bind(this) });
+
+    // User registration form
+    this.registerForm = new FormGroup({
+      userEmailControl: this.userEmailControl,
+      userPasswordsGroup: this.userPasswordsGroup
+    });
+    // this.registerForm.setValidators(this.passwordsValidator);
+  }
 
   ngOnInit(): void {
+
   }
 
-  // fake validation of confirm passwords
-  userPasswordIsValid(): boolean {
-    return true;
+  passwordsValid(control: AbstractControl): { [key: string]: boolean } | null {
+    this.errorMessage = '';
+    const password1 = control.get('userPasswordControl')?.value.toString();
+    const password2 = control.get('userPasswordConfirmControl')?.value.toString();
+    
+    if(password1.indexOf(' ') >= 0) {
```

Comment:
use .includes(' ') or .contains, whatever the javascript / typescript method is. a lot cleaner and more understandable